usecases/monitoring/grpc.go

```diff
@@ -40,13 +40,13 @@ const (
 //
 // ```
 //
-//	svrMetrics := monitoring.NewServerMetrics(metrics, prometheus.DefaultRegisterer)
+//	svrMetrics := monitoring.NewGRPCServerMetrics(metrics, prometheus.DefaultRegisterer)
 //	grpcServer := grpc.NewServer(monitoring.InstrumentGrpc(*svrMetrics)...)
 //
 //	grpcServer.Serve(listener)
 //
 // ```
-func InstrumentGrpc(svrMetrics ServerMetrics) []grpc.ServerOption {
+func InstrumentGrpc(svrMetrics *ServerMetrics) []grpc.ServerOption {
```

Comment:
Have you considered adding industry-standard HTTP and gRPC metrics like:
* `http_requests_total{method="GET", status="200"}`
* `grpc_server_handled_total{grpc_service="UserService", grpc_method="GetUser", grpc_code="OK"}`

There are good go libs for that:
1. https://github.com/slok/go-http-metrics
2. https://github.com/grpc-ecosystem/go-grpc-middleware

Reply:
Hmm. good point. I agree. may need bit of refactoring. But I think worth doing at this stage. I love the separation.

Reply:
Done.